src/index.js

```diff
@@ -110,11 +113,15 @@ function fetchConfigFromReq({ network, req }) {
     fetchParams.method = 'POST'
     fetchParams.headers = {
       'Accept': 'application/json',
-      'Content-Type': 'application/json'
+      'Content-Type': 'application/json',
+      [source]: requestOrigin
```

Comment:
I guess my big question about adding a new header: Does this cause a pre-flight, doubling our load on Infura? I'm not a header pre-flight expert, that would really be my top question for @kumavis.

Reply:
@jennypollack @danfinlay  if you inspect the network requests on the BG and you see a bunch of OPTIONS requests right after the POST then the answer is yes.

Reply:
just spent some time clocking network activity between my branch and the current release (5.3.1) for mainnet and rinkeby and the numbers don't lead me to believe that there is a pre-flight. i'm also not seeing any OPTIONS requests before the POST as @brunobar79 mentioned